ui/src/widgets/ui-gauge/UIGauge.vue

```diff
@@ -32,12 +30,74 @@ export default {
         value: function () {
             return this.messages[this.id]?.payload
         },
+        title () {
+            return this.getProperty('title')
+        },
+        gtype () {
+            return this.getProperty('gtype')
+        },
+        gstyle () {
+            return this.getProperty('gstyle')
+        },
+        prefix () {
+            return this.getProperty('prefix')
+        },
+        suffix () {
+            return this.getProperty('suffix')
+        },
+        units () {
+            return this.getProperty('units')
+        },
         icon () {
-            return this.props.icon?.replace(/^mdi-/, '')
```

Comment:
Replaced `return this.props.icon?.replace(/^mdi-/, '')` from `return this.getProperty('icon')` as material icon maker is not being used

Reply:
> as material icon maker is not being used

It is. Users can pass in `mdi-<icon>` _or_ `<icon>` directly and we should support that

Reply:
@joepavitt @cstns 
To be more clearer this dynamic icon only supports "Half Gauge" and "3/4 Gauge" and the functionality is already implemented in `UIGaugeDialog.vue` which is a child component comes under the `UIGauage.vue` therefore I used `icon ()` computed property to fetch dynamic icon changes

<img width="514" alt="Screenshot 2024-09-03 at 13 58 45" src="https://github.com/user-attachments/assets/5fe230ad-d5eb-4e77-a166-383989d573fa">